### PR TITLE
Add readonly field for MongoDB

### PR DESCRIPTION
**Pull request by zbarahal-do**
Support for creating MongoDB read only users - merge is dependent on feature flag enablement.

### CODE REVIEW

**Review by dbrian57 — APPROVED** [review]
@andrewsomething  Hey, this looks fine to me, but is the `Add New Read Only User (MongoDB Only)` line actually gonna show up in the docs? The `Add New User with MySQL Auth Plugin` doesn't show up as far as I can tell. Just worried about making it confusing.

**Review by ChiefMateStarbuck — APPROVED** [review]

**Comment by danaelhe** [comment]
> @andrewsomething Hey, this looks fine to me, but is the `Add New Read Only User (MongoDB Only)` line actually gonna show up in the docs? The `Add New User with MySQL Auth Plugin` doesn't show up as far as I can tell. Just worried about making it confusing.

I do see the `Add New User with MySQL Auth Plugin` in the dropdown of the request samples. Are you expecting to see it somewhere else?

File: specification/resources/databases/databases_add_user.yml [hunk 1]
```diff
@@ -38,6 +38,11 @@ requestBody:
             mysql_settings:
               auth_plugin: mysql_native_password
 
+        Add New Read Only User (MongoDB Only):
+          value:
+            name: my-readonly
+            readonly: true
```
Comment (danaelhe) [inline specification/resources/databases/databases_add_user.yml:44]: I see the `Add New User with MySQL Auth Plugin` and `Add New User` are also documented in the example responsebody [here](https://github.com/digitalocean/openapi/blob/main/specification/resources/databases/responses/user.yml#L28)

Might be worth adding the `Add New Read Only User (MongoDB Only)` example responsebody as well.

**Review by danaelhe — COMMENTED** [review]
Looks good! Have one small suggestion.

**Pull request merged**